ExecutionProperty is not working in GStrings

For DSL

```groovy
io.codearte.accurest.dsl.GroovyDsl.make {
                request {
                    method """PUT"""
                    url """/fraudcheck"""
                    body("""
                        {
                        "clientPesel":"${value(client(regex('[0-9]{10}')), server('1234567890'))}",
                        "loanAmount":123.123
                        }
                    """
                    )
                    headers {
                        header("""Content-Type""", """application/vnd.fraud.v1+json""")

                    }
                
                }
            response {
                status 200
                body( """{
    "fraudCheckStatus": "OK",
    "rejectionReason": ${value(client(null), server(execute('assertThatRejectionReasonIsNull($it)')))}
}""")
                headers {
                     header('Content-Type': 'application/vnd.fraud.v1+json')

                    }
                
            }
        
}
```

an exception is thrown while trying to generate Wiremock stubs

```
      at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
        ... 57 more
Caused by: groovy.json.JsonException: Unable to determine the current character, it is not a string, number, array, or object

The current character read is 'i' with an int value of 105
Unable to determine the current character, it is not a string, number, array, or object
line number 3
index number 55
    "rejectionReason": io.codearte.accurest.dsl.internal.ExecutionProperty@26d24d7a
.......................^
        at groovy.json.internal.JsonParserCharArray.decodeValueInternal(JsonParserCharArray.java:216)
        at groovy.json.internal.JsonParserCharArray.decodeJsonObject(JsonParserCharArray.java:140)
        at groovy.json.internal.JsonParserCharArray.decodeValueInternal(JsonParserCharArray.java:196)
        at groovy.json.internal.JsonParserCharArray.decodeValue(JsonParserCharArray.java:166)
        at groovy.json.internal.JsonParserCharArray.decodeFromChars(JsonParserCharArray.java:45)
        at groovy.json.internal.JsonParserCharArray.parse(JsonParserCharArray.java:409)
        at groovy.json.internal.BaseJsonParser.parse(BaseJsonParser.java:103)
        at groovy.json.JsonSlurper.parseText(JsonSlurper.java:208)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoCachedMethodSiteNoUnwrap.invoke(PojoMetaMethodSite.java:210)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:53)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at io.codearte.accurest.dsl.internal.Body.extractValue(Body.groovy:63)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
        at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite$StaticMetaMethodSiteNoUnwrapNoCoerce.invoke(StaticMetaMethodSite.java:148)
        at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite.callStatic(StaticMetaMethodSite.java:99)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:169)
        at io.codearte.accurest.dsl.internal.Body.<init>(Body.groovy:38)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
```

The problem is the same as with `Pattern`. I guess more hacks will have to be added to the `io.codearte.accurest.dsl.internal.Body#extractValue(groovy.lang.GString, groovy.lang.Closure)` ;)

--- a/accurest-core/src/main/groovy/io/codearte/accurest/util/ContentUtils.groovy
+++ b/accurest-core/src/main/groovy/io/codearte/accurest/util/ContentUtils.groovy
@@ -179,6 +179,30 @@ class ContentUtils {
 		})
 	}
 
+	/**
+	 * <p>
+	 *     If you wonder why there is val[1] without null-check then take a look at this:
+	 * </p>
+	 * <p>
+	 *     Example:
+	 * </p>
+	 * <p>
+	 *     Our string equals: {@code EXECUTION>>assertThatRejectionReasonIsNull($it)<<}
+	 *     The matcher matches this group with the pattern {@code EXECUTION>>(.*)<<}
+	 * </p>
+	 * <p>
+	 * So {@code executionMatcher[0]} returns 2 elements:
+	 *     <ul>
+	 *         <li> index0: EXECUTION>>assertThatRejectionReasonIsNull($it)<< </li>
+	 *         <li> index1: assertThatRejectionReasonIsNull($it)<< </li>
+	 *     </ul>
+	 * </p>
+	 * <p>
+	 *    Thus one can safely write {@code executionMatcher[0][1]} to retrieve the matched group
+	 * </p>
+	 * @param string to match the regexps against
+	 * @return object converted from temporary holders
+	 */
 	static Object returnParsedObject(String string) {
 		Matcher matcher = TEMPORARY_PATTERN_HOLDER.matcher(string.trim())
 		if (matcher.matches()) {

--- a/accurest-core/src/main/groovy/io/codearte/accurest/util/ContentUtils.groovy
+++ b/accurest-core/src/main/groovy/io/codearte/accurest/util/ContentUtils.groovy
@@ -5,6 +5,7 @@ import groovy.json.JsonSlurper
 import groovy.transform.TypeChecked
 import groovy.util.logging.Slf4j
 import io.codearte.accurest.dsl.internal.DslProperty
+import io.codearte.accurest.dsl.internal.ExecutionProperty
 import io.codearte.accurest.dsl.internal.Headers
 import io.codearte.accurest.dsl.internal.MatchingStrategy
 import org.codehaus.groovy.runtime.GStringImpl
@@ -24,7 +25,9 @@ class ContentUtils {
 	}
 
 	private static final Pattern TEMPORARY_PATTERN_HOLDER = Pattern.compile('REGEXP>>(.*)<<')
+	private static final Pattern TEMPORARY_EXECUTION_PATTERN_HOLDER = Pattern.compile('EXECUTION>>(.*)<<')
 	private static final String JSON_VALUE_PATTERN_FOR_REGEX = 'REGEXP>>%s<<'
+	private static final String JSON_VALUE_PATTERN_FOR_EXECUTION = '"EXECUTION>>%s<<"'
 
 	/**
 	 * Due to the fact that we allow users to have a body with GString and different values inside
@@ -154,6 +157,10 @@ class ContentUtils {
 		return String.format(JSON_VALUE_PATTERN_FOR_REGEX, pattern.pattern())
 	}
 
+	private static String transformJSONStringValue(ExecutionProperty property, Closure valueProvider) {
+		return String.format(JSON_VALUE_PATTERN_FOR_EXECUTION, property.executionCommand)
+	}
+
 	private static String transformXMLStringValue(Object obj, Closure valueProvider) {
 		return escapeXml11(obj.toString())
 	}
@@ -166,16 +173,26 @@ class ContentUtils {
 		MapConverter.transformValues(parsedJson, { Object value ->
 			if (value instanceof String) {
 				String string = (String) value
-				Matcher matcher = TEMPORARY_PATTERN_HOLDER.matcher(string.trim())
-				if (matcher.matches()) {
-					List val = matcher[0] as List
-					String pattern = val[1]
-					return Pattern.compile(pattern)
-				}
-				return value
+				return returnParsedObject(string)
 			}
 			return value
 		})
+	}
+
+	static Object returnParsedObject(String string) {
+		Matcher matcher = TEMPORARY_PATTERN_HOLDER.matcher(string.trim())
+		if (matcher.matches()) {
+			List val = matcher[0] as List
+			String pattern = val[1]
+			return Pattern.compile(pattern)
+		}
+		Matcher executionMatcher = TEMPORARY_EXECUTION_PATTERN_HOLDER.matcher(string.trim())
+		if (executionMatcher.matches()) {
+			List val = executionMatcher[0] as List
+			String pattern = val[1]
+			return new ExecutionProperty(pattern)
+		}
+		return string
 	}
 
 	public static ContentType recognizeContentTypeFromHeader(Headers headers) {

--- a/accurest-core/src/test/groovy/io/codearte/accurest/builder/MockMvcSpockMethodBuilderSpec.groovy
+++ b/accurest-core/src/test/groovy/io/codearte/accurest/builder/MockMvcSpockMethodBuilderSpec.groovy
@@ -551,4 +551,47 @@ class MockMvcSpockMethodBuilderSpec extends Specification implements WireMockStu
 		then:
 			spockTest.contains('''$[?(@.message =~ /User not found by email = \\\\[[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\\\.[a-zA-Z]{2,4}\\\\]/)]''')
 	}
+
+	@Issue('72')
+	def "should make the execute method work"() {
+		given:
+			GroovyDsl contractDsl = GroovyDsl.make {
+				request {
+					method """PUT"""
+					url """/fraudcheck"""
+					body("""
+                        {
+                        "clientPesel":"${value(client(regex('[0-9]{10}')), server('1234567890'))}",
+                        "loanAmount":123.123
+                        }
+                    """
+					)
+					headers {
+						header("""Content-Type""", """application/vnd.fraud.v1+json""")
+
+					}
+
+				}
+				response {
+					status 200
+					body( """{
+    "fraudCheckStatus": "OK",
+    "rejectionReason": ${value(client(null), server(execute('assertThatRejectionReasonIsNull($it)')))}
+}""")
+					headers {
+						header('Content-Type': 'application/vnd.fraud.v1+json')
+
+					}
+
+				}
+
+			}
+			MockMvcSpockMethodBodyBuilder builder = new MockMvcSpockMethodBodyBuilder(contractDsl)
+			BlockBuilder blockBuilder = new BlockBuilder(" ")
+		when:
+			builder.appendTo(blockBuilder)
+			def spockTest = blockBuilder.toString()
+		then:
+			spockTest.contains('''assertThatRejectionReasonIsNull(parsedJson.read('$.rejectionReason'))''')
+	}
 }

